### PR TITLE
[WIP] Batch messages sent over DDP from server to client

### PR DESCRIPTION
**Pull request by KoenLav**
Messages sent from the client to the server were already batched, similarly. Hence I copied the code which was used to batch messages on the client and applied it in a similar manner on the server.

Also added a TODO on the seemingly unneeded use of Fibers in processMessage.

Requesting feedback!

### CODE REVIEW

**Comment by sebakerckhof** [comment]
note that some people use alternative ddp clients like:
- https://github.com/mondora/asteroid
- https://github.com/oortcloud/node-ddp-client
- https://github.com/sonyarouje/DDPClient.NET

This would break those clients since they would not be able to handle the incoming arrays.

So this should at least be opt-in behind a setting or something IMO.

**Comment by KoenLav** [comment]
@sebakerckhof agreed!

I figured imposing this delay without making it configurable would be a bad way to go either way.

**Comment by mitar** [comment]
I thought we are already batching messages from server to client, on client, before passing them on to app code.

**Comment by KoenLav** [comment]
@mitar indeed, this does already happen on the client.

However, this patch aims to address problems with the server ending up with high load when a lot of DDP messages need to be sent to a lot of clients.

File: packages/ddp-client/test/livedata_connection_tests.js [hunk 1]
```diff
@@ -1842,7 +1842,8 @@ _.each(DDPCommon.SUPPORTED_DDP_VERSIONS, function(version) {
         test.equal(connection._version, version);
         // It's a little naughty to access _stream and _send, but it works...
         connection._stream.on('message', function(json) {
-          var msg = JSON.parse(json);
+          var messages = JSON.parse(json);
+          var msg = messages.constructor === Array ? messages[0] : messages;
```
Comment (copleykj) [inline packages/ddp-client/test/livedata_connection_tests.js]: use instanceof here as  well

File: packages/ddp-client/common/livedata_connection.js [hunk 2]
```diff
@@ -1629,47 +1629,54 @@ export class Connection {
...
-    if (msg === null || !msg.msg) {
-      // XXX COMPAT WITH 0.6.6. ignore the old welcome message for back
-      // compat.  Remove this 'if' once the server stops sending welcome
-      // messages (stream_server.js).
-      if (!(msg && msg.server_id))
-        Meteor._debug('discarding invalid livedata message', msg);
-      return;
-    }
+    // Convert all DDP messages to Array form
+    messages = messages.constructor === Array ? messages : [messages];
```
[the remark this answers is not in the document]
Comment (copleykj) [reply packages/ddp-client/common/livedata_connection.js]: `Array.isArray` would also suffice

**Comment by sebakerckhof** [comment]
@copleykj : Wrong PR, this one is closed and followed up by: https://github.com/meteor/meteor/pull/9862

**Comment by copleykj** [comment]
Well WTF :) I swear I came to this through the PR tab of open PR's

**Pull request closed without merging**